reactor-core/src/test/java/reactor/core/publisher/SinkManyBestEffortTest.java

```diff
@@ -0,0 +1,369 @@
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+		sink.tryEmitComplete().orThrow();
+
+		assertThat(sink.tryEmitComplete()).as("tryEmitComplete").isEqualTo(Emission.FAIL_TERMINATED);
+	}
+
+	@Test
+	void tryEmitErrorNoSubscribersRetains() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		assertThat(sink.tryEmitError(new IllegalStateException("boom"))).as("tryEmitError").isEqualTo(Emission.OK);
+
+		StepVerifier.create(sink.asFlux())
+		            .verifyErrorMessage("boom");
+	}
+
+	@Test
+	void tryEmitErrorOnTerminated() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+		sink.tryEmitError(new IllegalStateException("boom")).orThrow();
+
+		assertThat(sink.tryEmitError(new IllegalStateException("boom ignored"))).as("tryEmitError").isEqualTo(Emission.FAIL_TERMINATED);
+
+		StepVerifier.create(sink.asFlux())
+		            .verifyErrorMessage("boom");
+	}
+
+	@Test
+	void tryEmitErrorOnSubscribers() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		AssertSubscriber<Integer> sub1 = AssertSubscriber.create();
+		AssertSubscriber<Integer> sub2 = AssertSubscriber.create(0);
+		sink.subscribe(sub1);
+		sink.subscribe(sub2);
+
+		assertThat(sink.tryEmitError(new IllegalStateException("boom"))).as("tryEmitError").isEqualTo(Emission.OK);
+
+		sub1.assertErrorMessage("boom");
+		sub2.assertErrorMessage("boom");
+	}
+
+	@Test
+	void addSubscriberThatCancelsOnSubscription() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		sink.subscribe(new BaseSubscriber<Integer>() {
+			@Override
+			protected void hookOnSubscribe(Subscription subscription) {
+				subscription.cancel();
+			}
+		});
+	}
+
+	@Test
+	void innersReflectsSubscribers() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		assertThat(sink.inners()).as("before subscribers").isEmpty();
+
+		Disposable sub1 = sink.subscribe();
+		Disposable sub2 = sink.subscribe();
+
+		assertThat(sink.inners()).hasSize(2);
+		assertThat(sink.inners().map(inner -> inner.scanUnsafe(Scannable.Attr.ACTUAL)))
+				.containsExactly(sub1, sub2);
+	}
+
+	@Test
+	void scanSink() {
+		SinkManyBestEffort<Integer> sinkNormal = SinkManyBestEffort.createBestEffort();
+
+		assertThat(sinkNormal.scan(Scannable.Attr.TERMINATED)).as("normal not terminated").isFalse();
+		sinkNormal.tryEmitComplete().orThrow();
+		assertThat(sinkNormal.scan(Scannable.Attr.TERMINATED)).as("normal terminated").isTrue();
+
+		SinkManyBestEffort<Integer> sinkError = SinkManyBestEffort.createBestEffort();
+		Throwable expectedError = new IllegalStateException("boom");
+		sinkError.tryEmitError(expectedError).orThrow();
+
+		assertThat(sinkError.scan(Scannable.Attr.TERMINATED)).as("error terminated").isTrue();
+		assertThat(sinkError.scan(Scannable.Attr.ERROR)).as("error captured").isSameAs(expectedError);
+	}
+
+	@Nested
+	class RaceConditionsAndLoops {
```

Comment:
I get a feeling that we will never start using JCstress if we will keep adding these for-based ones :( 
this is a new code, goes into master where JCstress is configured, and I see no reason not to use it.

Reply:
I have no idea what I'm doing, but done